pkg/component/nodemanagement/machinecontrollermanager/machine_controller_manager.go

```diff
@@ -320,78 +333,105 @@ func (m *machineControllerManager) Deploy(ctx context.Context) error {
+					RelabelConfigs: []monitoringv1.RelabelConfig{{
+						Action: "labelmap",
+						Regex:  `__meta_kubernetes_service_label_(.+)`,
+					}},
+					MetricRelabelConfigs: monitoringutils.StandardMetricRelabelConfig(
+						// Machine Deployment related metrics
+						"mcm_machine_deployment_items_total",
+						"mcm_machine_deployment_info",
+						"mcm_machine_deployment_info_spec_paused",
+						"mcm_machine_deployment_info_spec_replicas",
+						"mcm_machine_deployment_info_spec_min_ready_seconds",
+						"mcm_machine_deployment_info_spec_rolling_update_max_surge",
+						"mcm_machine_deployment_info_spec_rolling_update_max_unavailable",
+						"mcm_machine_deployment_info_spec_revision_history_limit",
+						"mcm_machine_deployment_info_spec_progress_deadline_seconds",
+						"mcm_machine_deployment_info_spec_rollback_to_revision",
+						"mcm_machine_deployment_status_condition",
+						"mcm_machine_deployment_status_available_replicas",
+						"mcm_machine_deployment_status_unavailable_replicas",
+						"mcm_machine_deployment_status_ready_replicas",
+						"mcm_machine_deployment_status_updated_replicas",
+						"mcm_machine_deployment_status_collision_count",
+						"mcm_machine_deployment_status_replicas",
+						"mcm_machine_deployment_failed_machines",
+						// Machine Set related metrics
+						"mcm_machine_set_items_total",
+						"mcm_machine_set_info",
+						"mcm_machine_set_failed_machines",
+						"mcm_machine_set_info_spec_replicas",
+						"mcm_machine_set_info_spec_min_ready_seconds",
+						"mcm_machine_set_status_condition",
+						"mcm_machine_set_status_available_replicas",
+						"mcm_machine_set_status_fully_labelled_replicas",
+						"mcm_machine_set_status_replicas",
+						"mcm_machine_set_status_ready_replicas",
+						// Machine related metrics
+						"mcm_machine_stale_machines_total",
+						// misc metrics
+						"mcm_misc_scrape_failure_total",
+						"process_max_fds",
+						"process_open_fds",
+						// workqueue related metrics
+						"mcm_workqueue_adds_total",
+						"mcm_workqueue_depth",
+						"mcm_workqueue_queue_duration_seconds_bucket",
+						"mcm_workqueue_queue_duration_seconds_sum",
+						"mcm_workqueue_queue_duration_seconds_count",
+						"mcm_workqueue_work_duration_seconds_bucket",
+						"mcm_workqueue_work_duration_seconds_sum",
+						"mcm_workqueue_work_duration_seconds_count",
+						"mcm_workqueue_unfinished_work_seconds",
+						"mcm_workqueue_longest_running_processor_seconds",
+						"mcm_workqueue_retries_total",
+					),
+				},
+				{
+					Port: portNameProviderMetrics,
+					RelabelConfigs: []monitoringv1.RelabelConfig{{
+						Action: "labelmap",
+						Regex:  `__meta_kubernetes_service_label_(.+)`,
+					}},
+					MetricRelabelConfigs: monitoringutils.StandardMetricRelabelConfig(
+						// Machine related metrics
+						"mcm_machine_items_total",
+						"mcm_machine_current_status_phase",
+						"mcm_machine_info",
+						"mcm_machine_status_condition",
```

Comment:
Why have the machine related metrics moved to this place? I compared both for an AWS cluster and they seem considerably different. Is this intended?

Example for `mcm_machine_status_condition` metric in cluster with 4 machines:

MCM container:
```
# HELP mcm_machine_status_condition Information of the mcm managed Machines' status conditions.
# TYPE mcm_machine_status_condition gauge
mcm_machine_status_condition{condition="ClusterNetworkProblem",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ClusterNetworkProblem",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ClusterNetworkProblem",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ClusterNetworkProblem",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="CorruptDockerOverlay2",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="CorruptDockerOverlay2",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="CorruptDockerOverlay2",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="CorruptDockerOverlay2",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="DiskPressure",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="DiskPressure",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="DiskPressure",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="DiskPressure",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentContainerdRestart",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentContainerdRestart",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentContainerdRestart",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentContainerdRestart",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentDockerRestart",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentDockerRestart",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentDockerRestart",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentDockerRestart",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentKubeletRestart",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentKubeletRestart",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentKubeletRestart",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentKubeletRestart",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentUnregisterNetDevice",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentUnregisterNetDevice",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentUnregisterNetDevice",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentUnregisterNetDevice",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="HostNetworkProblem",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="HostNetworkProblem",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="HostNetworkProblem",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="HostNetworkProblem",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="KernelDeadlock",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="KernelDeadlock",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="KernelDeadlock",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="KernelDeadlock",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="MemoryPressure",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="MemoryPressure",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="MemoryPressure",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="MemoryPressure",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="PIDPressure",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="PIDPressure",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="PIDPressure",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="PIDPressure",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ReadonlyFilesystem",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ReadonlyFilesystem",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ReadonlyFilesystem",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="ReadonlyFilesystem",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="Ready",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 1
mcm_machine_status_condition{condition="Ready",name="shoot--foo--bar-p2-z1-67497-hb2tk",namespace="shoot--foo--bar"} 1
mcm_machine_status_condition{condition="Ready",name="shoot--foo--bar-p2-z2-86867-wzpg6",namespace="shoot--foo--bar"} 1
mcm_machine_status_condition{condition="Ready",name="shoot--foo--bar-p3-z1-6b74f-fs7fd",namespace="shoot--foo--bar"} 1
```

Provider container:
```
# HELP mcm_machine_status_condition Information of the mcm managed Machines' status conditions.
# TYPE mcm_machine_status_condition gauge
mcm_machine_status_condition{condition="ClusterNetworkProblem",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="FrequentContainerdRestart",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--foo--bar"} 0
mcm_machine_status_condition{condition="HostNetworkProblem",name="shoot--foo--bar-p1-z1-5c65d-j25cp",namespace="shoot--scalesim--scenario-c"} 0
```

Reply:
As confirmed by @aaronfern, machine related metrics were always exclusively exposed by the provider container. We took more samples to prove that, the one above is faulty and can be ignored.